akka-stream/src/main/scala/akka/stream/impl/FanIn.scala

```diff
@@ -155,6 +155,7 @@ private[akka] object FanIn {
       if (input.inputsDepleted) {
         if (marked(id)) markedDepleted += 1
         depleted(id) = true
+        onDepleted(id)
```

Comment:
This is killing FlexiMergeDocTest (of all things!), but validly so: eagerClose is now way too eager, leading to `onComplete` before the last element can be emitted. What was the rationale behind this change @drewhk ?